src/clusters.rs

```diff
@@ -0,0 +1,940 @@
+use std::f64::consts::PI;
+use rand::Rng;
+use rand::prelude::ThreadRng;
+use json::JsonValue;
+
+use crate::vector::Vec3;
+use crate::half_multivector::HalfMultivector;
+use crate::point::InternalPoint;
+
+/// This trait is used to arrange a set of points to represent an initial
+/// set that will be sent through a Chaos Game algorithm. Typically, this is
+/// done by randomly generating a set of points in some arrangement like a
+/// box or line.
+pub trait Cluster {
+    /// Generate a set of points. This may be called several times, and each
+    /// time it must produce a new set of points.
+    fn generate(&mut self, set_id: u16) -> Vec<InternalPoint>; 
+    /// Get the number of points in the initial set for measuring complexity.
+    fn len(&self) -> usize;
+}
+
+pub struct Points {
```

Comment:
document!

(😆 what if there was a `document!()` macro that did the documentation for you?)